src/Http/Http.Features/src/FeatureReferences.cs

```diff
@@ -61,10 +61,10 @@ public void Initalize(IFeatureCollection collection, int revision)
         //
         // Generally Fetch is called at a ratio > x4 of UpdateCached so this is a large gain
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public TFeature Fetch<TFeature, TState>(
-            [AllowNull, MaybeNull] ref TFeature cached,
+        public TFeature? Fetch<TFeature, TState>(
+            ref TFeature? cached,
             TState state,
-            Func<TState, TFeature> factory) where TFeature : class?
+            Func<TState, TFeature?> factory) where TFeature : class
```

Comment:
📝 There are some suppressions (`!` operators) in the implementation of this method which can now be removed.